### ISSUE TITLE
[Feature request] Option to ignore paths from being validate

### ISSUE THREAD

**Issue by vasfvitor** [HiDeoo/starlight-links-validator#31]
### Is your feature request related to a problem?

Pages with `.` doesn't work directly in Starlight/Astro, but utilizing the `slug` makes it work. Either way, the links validator fails for those links, even when the page is navigable.

### Describe the solution you'd like

An option to ignore a path/pages from being checked for links.

### Additional Context

Context: Pages are software release versions, hence the dots (`/release/@pkg/v.1.20`). Not sure `@` would interfere.

Thank you and I can contribute, if you accept this feature. Just give me a general direction or specification of how you want it

**Comment by HiDeoo** [comment]
Thanks for your feedback :raised_hands: 

This took a bit of a tangent from the original proposal as I have to say, having an option to ignore some links/paths is not something I'm a huge fan of at first glance.

While looking into the issue, I discovered that pages with a custom slug were not handled at all and after prototyping a solution, it turns out that this is something the validator can support and actually properly validate theses pages.
This also removes the need for an option to ignore some links/paths.

I just released [`v0.6.0`](https://github.com/HiDeoo/starlight-links-validator/releases/tag/v0.6.0) which should fix your issue.

I also made sure to add tests using a custom slug similar to what you're using with `.` and `@` to make sure it does not error for valid custom slugs but does error if the custom slug is invalid or forgotten altogether.

```md
---
# src/content/docs/guides/content.md
title: Page with custom slug
slug: release/@pkg/v0.1.0
---
```

```md
---
# src/content/docs/guides/test.md
title: Tests
---

- [A link](/release/@pkg/v0.1.0) <!-- this will be valid -->
- [A link](/release/@pkg/v0.2.0) <!-- this will be invalid -->
- [A link](/guides/content) <!-- this will be invalid -->
```

**Comment by vasfvitor** [comment]
THANK YOU SO MUCH THIS IS PERFECT!
